### PR TITLE
Add signup functionality

### PR DESCRIPTION
**Pull request by blakes24**
### What does this PR do?
- This PR allows users to sign up

### Description of Task to be completed
- A user enters their username, email, and password to sign up.

### How should this be manually tested?
- Clone the repository
- Access the signup form at `http://localhost:5000/signup`

### CODE REVIEW

**Comment by alexkayabula** [comment]
@blakes24 thank you for the Pull Request but the pull request does not have a description and does not show how to test the functionality. Use this [example](https://github.com/alexkayabula/fast-food-fast-react/pull/13) as reference on how to structure a pull request.

File: templates/base.html [hunk 1]
```diff
@@ -11,6 +11,16 @@
...
+    {% else %}
+        {% include 'nav-logged-out.html' %}
+    {% endif %}
+
+    {% for category, message in get_flashed_messages(with_categories=True) %}
+    <div class="alert alert-{{ category }}">{{ message }}</div>
+    {% endfor %}
+
     {% block content %}
     {% endblock %}
```
Comment (alexkayabula) [inline templates/base.html:26]: This implementation should not be part of the back-end implementation. Kindly remove it and implement it when working on the front-end of the application. I advise that you work on the back-end features first then work on the front-end after all back-end features are complete.

File: templates/index.html [hunk 2]
```diff
@@ -1,8 +1,8 @@
 {% extends 'base.html' %}
 
-{% include 'nav-logged-out.html' %}
-
 {% block content %}
+<h1 class="text-center">Dive Alas</h1>
+<p class="text-center">Use the search bar or drop a pin on the map to find nearby dive sites.</p>
 <div class="container text-center">
     <form class="form-inline justify-content-center" id="search-form">
         <input class="form-control" type="search" placeholder="Search" aria-label="Search" id="search-text">
```
Comment (alexkayabula) [inline templates/index.html:8]: This implementation should not be part of the back-end implementation. Kindly remove it and implement it when working on the front-end of the application. I advise that you work on the back-end features first then work on the front-end after all backend features are complete.

File: templates/nav-logged-in.html [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+            <li class="nav-item">
+                <a class="nav-link" href="/journal">Dive Journal</a>
+            </li>
+            <li class="nav-item">
+                <a class="nav-link" href="/logout">Logout</a>
+            </li>
+        </ul>
+    </div>
+    <a class="nav-item" href="#">{{g.user.username}}</a>
+</nav>
```
Comment (alexkayabula) [inline templates/nav-logged-in.html:22]: This implementation should not be part of the back-end implementation. Kindly remove it and implement it when working on the front-end of the application. I advise that you work on the back-end features first then work on the front-end after all backend features are complete.

File: templates/nav-logged-out.html [hunk 4]
```diff
@@ -1,17 +1,17 @@
...
-                <a class="nav-link" href="#">Home <span class="sr-only">(current)</span></a>
+            <li class="nav-item">
+                <a class="nav-link" href="/signup">Sign Up</a>
             </li>
             <li class="nav-item">
-                <a class="nav-link" href="#">Sign Up</a>
+                <a class="nav-link" href="/sign-in">Sign In</a>
             </li>
         </ul>
     </div>
```
Comment (alexkayabula) [inline templates/nav-logged-out.html:17]: This implementation should not be part of the back-end implementation. Kindly remove it and implement it when working on the front-end of the application. I advise that you work on the back-end features first then work on the front-end after all backend features are complete.

File: templates/signup.html [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+        {% endfor %}
+        {{ field(placeholder=field.label.text, class="form-control") }}
+      {% endfor %}
+
+      <button class="btn btn-primary">Sign up!</button>
+    </form>
+  </div>
+</div>
+
+{% endblock %}
```
Comment (alexkayabula) [inline templates/signup.html:24]: This implementation should not be part of the back-end implementation. Kindly remove it and implement it when working on the front-end of the application. I advise that you work on the back-end features first then work on the front-end after all backend features are complete.

File: templates/base.html (hunk 1 above)
Comment (blakes24) [reply templates/base.html:26]: agreed not to separate front and back end implementations

File: templates/index.html (hunk 2 above)
Comment (blakes24) [reply templates/index.html:8]: Changes were not necessary.

File: templates/nav-logged-in.html (hunk 3 above)
Comment (blakes24) [reply templates/nav-logged-in.html:22]: Changes are not required.

File: templates/nav-logged-out.html (hunk 4 above)
Comment (blakes24) [reply templates/nav-logged-out.html:17]: Changes are not required.

File: templates/signup.html (hunk 5 above)
Comment (blakes24) [reply templates/signup.html:24]: Changes are not required.

**Comment by alexkayabula** [comment]
🙂🙂.......Merged

**Pull request merged**